sdk/identity/Azure.Identity/src/AuthorizationCodeCredential.cs

```diff
@@ -18,12 +17,13 @@ namespace Azure.Identity
     /// </summary>
     public class AuthorizationCodeCredential : TokenCredential
     {
-        private readonly IConfidentialClientApplication _confidentialClient;
-        private readonly ClientDiagnostics _clientDiagnostics;
         private readonly string _authCode;
         private readonly string _clientId;
         private readonly CredentialPipeline _pipeline;
         private AuthenticationRecord _record;
+        private readonly string _tenantId;
+        private readonly TokenCredentialOptions _options;
```

Comment:
Generally we only read the options in the constructor and cache the fields rather than caching the entire options object. This way if the values of the options object change after construction the credential behavior doesn't change.

Reply:
I mainly did this so that the options could be passed to the `TenantIdResolver`. I could also just save off the `PreferClientConfiguredTenantId` property on the options, but I thought this allowed the logic in the resolver to stay a bit more encapsulated.

Reply:
I think we should capture `AllowMultiTenantAuthentication` from the options to avoid changing behavior after the credential is constructed.